### ISSUE TITLE
iex is started after my application

### ISSUE THREAD

**Issue by fuelen** [elixir-lang/elixir#13442]
### Elixir and Erlang/OTP versions

Elixir 1.16.2
OTP: 26.2.3

### Operating system

Linux

### Current behavior

I have a snippet of code which doesn't start background processing if iex is enabled in order not to pollute output with logs.
Check:
```elixir
  defp iex_enabled? do
    Code.ensure_loaded?(IEx) and IEx.started?()
  end
```
After upgrade to Elixir 1.16.2 and OTP 26.2.3 this code sometimes returns true, sometimes returns false.
```sh
$ iex -S mix
[(my_app 1.0.0) lib/my_app/application.ex:42: MyApp.Application.oban_config/0]
iex_enabled?() #=> false
```

### Expected behavior

Start iex before my application. It used to work like this with
```
erlang 25.1.2
elixir 1.14.2-otp-25
```

**Comment by josevalim** [comment]
Can you please try main?

**Comment by fuelen** [comment]
Works fine with main

**Comment by josevalim** [comment]
Alright, it will be fixed on 1.17 then. Unfortunately it is not a straight-forward change to backport.
